astroquery/gemini/core.py

```diff
@@ -0,0 +1,480 @@
+        if radius is not None:
+            if isinstance(radius, (int, float)):
+                radius = radius * units.deg
+            radius = astropy.coordinates.Angle(radius)
+
+        url = "%s/jsonsummary/notengineering/NotFail" % self.server
+        if coordinates is not None:
+            coordinates = commons.parse_coordinates(coordinates)
+            url = "%s/ra=%f/dec=%f" % (url, coordinates.ra.deg, coordinates.dec.deg)
+        if radius is not None:
+            url = "%s/sr=%fd" % (url, radius.deg)
+        if pi_name is not None:
+            url = "%s/PIname=%s" % (url, pi_name)
+        if program_id is not None:
+            url = "%s/progid=%s" % (url, program_id.upper())
+        if utc_date is not None:
+            if isinstance(utc_date, date):
+                url = "%s/%s" % (url, utc_date.strftime("YYYYMMdd"))
+            elif isinstance(utc_date, tuple):
+                if len(utc_date) != 2:
+                    raise ValueError("utc_date tuple should have two values")
+                if not isinstance(utc_date[0], date) or not isinstance(utc_date[1], date):
+                    raise ValueError("utc_date tuple should have date values in it")
+                url = "%s/%s-%s" % (url, utc_date[0].strftime("YYYYMMdd"), utc_date[1].strftime("YYYYMMdd"))
+        if instrument is not None:
+            if instrument.upper() not in __valid_instruments__:
+                raise ValueError("Unrecognized instrument: %s" % instrument)
+            url = "%s/%s" % (url, instrument)
+        if observation_class is not None:
+            if observation_class not in __valid_observation_class__:
+                raise ValueError("Unrecognized observation class: %s" % observation_class)
+            url = "%s/%s" % (url, observation_class)
+        if observation_type is not None:
+            if observation_type not in __valid_observation_types__:
+                raise ValueError("Unrecognized observation type: %s" % observation_type)
+            url = "%s/%s" % (url, observation_type)
+        if mode is not None:
+            if mode not in __valid_modes__:
+                raise ValueError("Unrecognized mode: %s" % mode)
+            url = "%s/%s" % mode
+        if adaptive_optics is not None:
+            if adaptive_optics not in __valid_adaptive_optics__:
+                raise ValueError("Unrecognized adaptive optics: %s" % adaptive_optics)
+            url = "%s/%s" % (url, adaptive_optics)
+        if program_text is not None:
+            url = "%s/ProgramText=%s" % (url, program_text)
+        if objectname is not None:
+            url = "%s/object=%s" % (url, objectname)
+        if raw_reduced is not None:
+            if raw_reduced not in __valid_raw_reduced__:
+                raise ValueError("Unrecognized raw/reduced setting: %s" % raw_reduced)
+            url = "%s/%s" % (url, raw_reduced)
+        response = self._request(method="GET", url=url, data={}, timeout=180, cache=False)
```

Comment:
`timeout` should be `self.TIMEOUT` and should be configurable, with a default to 180

`cache` should _not_ be hard-coded to `False` here, but should be a user parameter, unless there is very good reason.  If there is such a reason, please add it here.